### ISSUE TITLE
Better indentation of strings in assertion messages containing multi-line expected/actual

### ISSUE THREAD

**Issue by Bananeweizen** [assertj/assertj#3167]
#### Feature summary
Some of the String assertion messages indent both the actual and expected value for readability with 2 spaces, for example
https://github.com/assertj/assertj/blob/d02a738d7522dc58c54ca1b782fa50db9fcb67fc/assertj-core/src/main/java/org/assertj/core/error/ShouldBeEqualIgnoringWhitespace.java#L34

If actual and expected consist of multiple lines of text, that indentation can lead to more confusion than being helpful, because only the first line is indented, the others are not. Example output:
```
// isEqualToIgnoringWhitespace
Expecting actual:
  "foo
bar
baz"
to be equal to:
  "moreFoo
bar
baz"
when ignoring whitespace differences
```

I think it would be beneficial to
* either indent or not indent _every_ line of actual and expected
* eventually unify the indentation usage in all String/Charsequence related assertions, since that indentation is currently hard-coded in each sub class of the base error message class. Also it seems the base error message class has some capabilities around multi-line arguments, which are not used in the sub classes.

**Comment by joel-costigliola** [comment]
I completely agree, that's something I had in mind but could not find to address, we already have done it for arrays and iterables btw

**Comment by CodingMasterLSW** [comment]
Hi @joel-costigliola, @scordio 

I've submitted a PR (#4108) this issue. Could you please take a look when you have a moment? I'd appreciate your feedback. Thanks!

**Comment by CodingMasterLSW** [comment]
Hi @joel-costigliola,

I’ve submitted a PR (#4114) for this issue.

As I worked on it, I realized that the number of classes to be updated is quite large. To make the review process more manageable, I thought it might be better to split the task into multiple PRs. Does this approach work for you? If you prefer having everything in a single PR, I will consolidate the subsequent changes into the current one.

In the meantime, I have categorized the potential refactoring targets below. One thing I’m uncertain about is the `Pattern/Match group`. I wonder if they strictly require this refactoring, or if they should be left as they are. I’d appreciate your thoughts on this.
### Contain CharSequence 
- ShouldContainCharSequence
- ShouldContainCharSequenceOnlyOnce
- ShouldBeSubstring
- ShouldNotContainCharSequence
- ShouldContainSequenceOfCharSequence
- ShouldContainSubsequenceOfCharSequence
- ShouldNotContainSequenceOfCharSequence
- ShouldNotContainSubsequenceOfCharSequence

### StartWith/EndWith 
- ShouldStartWithIgnoringCase
- ShouldEndWithIgnoringCase
- ShouldNotStartWithIgnoringCase
- ShouldNotEndWithIgnoringCase

### Blank/Whitespace
- ShouldBeBlank
- ShouldNotBeBlank
- ShouldContainOnlyWhitespaces
- ShouldNotContainOnlyWhitespaces
- ShouldContainOneOrMoreWhitespaces
- ShouldNotContainAnyWhitespaces
- ShouldNotStartWithWhitespaces
- ShouldNotEndWithWhitespaces

### Pattern/Match 
- ShouldContainPattern
- ShouldNotContainPattern
- ShouldMatchPattern
- ShouldNotMatchPattern
- ShouldBeMixedCase
- ShouldContainOnlyDigits

Please let me know if you see any classes in the Pattern/Match group (or any others) that don’t need modification, or if there are any missing classes I should include.

Thanks!

**Comment by joel-costigliola** [comment]
I think we should also apply the indentation fix on the Pattern/Match error messages but only for the actual value, I don't think it would make sense to apply it a regex pattern (said otherwise let's wait for a user to ask for an indented pattern, until then we keep it as it is.)

**Comment by CodingMasterLSW** [comment]
Hi @joel-costigliola,

I’ve submitted a PR (https://github.com/assertj/assertj/pull/4122) for `StartWith/EndWith`

**Comment by CodingMasterLSW** [comment]
Hi @joel-costigliola,

I’ve submitted a PR (https://github.com/assertj/assertj/pull/4129) for `Blank/Whitespace`

I also included a question about `ShouldBeBlank` and `ShouldNotBeBlank` in the PR description. I would appreciate it if you could take a look and share your thoughts when you have a chance.🙂

**Comment by CodingMasterLSW** [comment]
Hi @joel-costigliola,

I’ve submitted a PR (https://github.com/assertj/assertj/pull/4133) for CharSequence and (https://github.com/assertj/assertj/pull/4134) pattern match. 

With these, it seems all the changes mentioned in the issue have been addressed. It’s been a pleasure working on this!
Please let me know if any further changes or adjustments are needed regarding this issue. 🙂
Cheers!

**Comment by joel-costigliola** [comment]
Since these PRs were merged, I'm closing this issue.

Thanks again for this chunky piece of work @CodingMasterLSW, much appreciated ! :ok_hand:
